### DISCUSSION TITLE
Get users from AD group

### DISCUSSION THREAD

**Discussion by cdias-sil** [YannickRe/spfx-msgraph-peoplesearch#70] [General]
Hi,

Is it possible to use this webpart to get users from specific or group of AD groups?

**Comment by YannickRe** [comment]
In the current form of this web part this isn't possible, unfortunately. It doesn't look to difficult to achieve this, for one specific group (set of Groups is not possible).

I haven't tested it, but I looked at the following docs: https://learn.microsoft.com/en-us/graph/api/group-list-members?view=graph-rest-1.0&tabs=http&WT.mc_id=M365-MVP-5003400#example-4-use-searchand-odata-cast-to-get-user-membership-in-groups-with-display-names-that-contain-the-letters-pr-including-a-count-of-returned-objects

As a result, I believe you will need to change the following line
https://github.com/YannickRe/spfx-msgraph-peoplesearch/blob/f1d014f55e1e1ae00f5d413bf074176456e0ab4e/src/services/SearchService/SearchService.ts#L41

to (replace {id} with the id of your group, of course)
```
.api('/groups/{id}/members/microsoft.graph.user')
```

**Reply by cdias-sil** [reply]
Thanks for your solution @YannickRe , it worked perfectly. Had to create an app for each group but for now that works for me.

I wish I understood how to configure webpart in code to allow multiple groups it is a shame it doesn't allow to add multiple or decide which group within the app settings.

**Comment by YannickRe** [comment]
Great to hear! The web part isn’t developed for this functionality, at the time of creation there was no api that supported this.
Same goes for multiple groups, there is no way to do this efficiently.
